extensions/hibernate-envers/runtime/src/main/java/io/quarkus/hibernate/envers/HibernateEnversBuildTimeConfig.java

```diff
@@ -43,4 +43,125 @@
     @ConfigItem(defaultValue = "REVTYPE")
     public Optional<String> revisionTypeFieldName;
 
+    /**
+     * Enable the revision_on_collection_change feature.
+     * Maps to {@link org.hibernate.envers.configuration.EnversSettings#REVISION_ON_COLLECTION_CHANGE}.
+     */
+    @ConfigItem(defaultValue = "true")
+    public boolean revisionOnCollectionChange;
+
+    /**
+     * Enable the do_not_audit_optimistic_locking_field feature.
+     * Maps to {@link org.hibernate.envers.configuration.EnversSettings#DO_NOT_AUDIT_OPTIMISTIC_LOCKING_FIELD}.
+     */
+    @ConfigItem(defaultValue = "true")
+    public boolean doNotAuditOptimisticLockingField;
+
+    /**
+     * Defines the default schema of where audit tables are to be created.
+     * Maps to {@link org.hibernate.envers.configuration.EnversSettings#DEFAULT_SCHEMA}.
+     */
+    @ConfigItem(defaultValue = "")
+    public Optional<String> defaultSchema;
+
+    /**
+     * Defines the default catalog of where audit tables are to be created.
+     * Maps to {@link org.hibernate.envers.configuration.EnversSettings#DEFAULT_CATALOG}.
+     */
+    @ConfigItem(defaultValue = "")
+    public Optional<String> defaultCatalog;
+
+    /**
+     * Enables the track_entities_changed_in_revision feature.
+     * Maps to {@link org.hibernate.envers.configuration.EnversSettings#TRACK_ENTITIES_CHANGED_IN_REVISION}.
+     */
+    @ConfigItem(defaultValue = "false")
+    public boolean trackEntitiesChangedInRevision;
+
+    /**
+     * Enables the use_revision_entity_with_native_id feature.
+     * Maps to {@link org.hibernate.envers.configuration.EnversSettings#USE_REVISION_ENTITY_WITH_NATIVE_ID}.
+     */
+    @ConfigItem(defaultValue = "true")
+    public boolean useRevisionEntityWithNativeId;
+
+    /**
+     * Enables the global_with_modified_flag feature.
+     * Maps to {@link org.hibernate.envers.configuration.EnversSettings#GLOBAL_WITH_MODIFIED_FLAG}.
+     */
+    @ConfigItem(defaultValue = "false")
+    public boolean globalWithModifiedFlag;
+
+    /**
+     * Defines the suffix to be used for modified flag columns. Defaults to {@literal _MOD}.
+     * Maps to {@link org.hibernate.envers.configuration.EnversSettings#MODIFIED_FLAG_SUFFIX}
+     */
+    @ConfigItem(defaultValue = "_MOD")
+    public Optional<String> modifiedFlagSuffix;
+
+    /**
+     * Defines the fully qualified class name of a user defined revision listener.
+     * Maps to {@link org.hibernate.envers.configuration.EnversSettings#REVISION_LISTENER}.
+     */
+    @ConfigItem
+    public Optional<String> revisionListener;
+
+    /**
+     * Defines the fully qualified class name of the audit strategy to be used.
+     * 
+     * Maps to {@link org.hibernate.envers.configuration.EnversSettings#AUDIT_STRATEGY}.
+     */
+    @ConfigItem(defaultValue = "org.hibernate.envers.strategy.DefaultAuditStrategy")
+    public Optional<String> auditStrategy;
```

Comment:
If you want this to work in native, you need to register this class for reflection.